frontend/toolkit/DataTable.vue

```diff
@@ -32,13 +31,14 @@
 
 <script>
   module.exports = {
-    props: [
-      "headers", "rows", "globalActions", "rowActions"
-    ],
+    props: {
+      headers: { type: Array, default: () => {return [];} },
+      columnFormatters: { type: Array, default: () => {return [];} },
+      rows: { type: Array, default: () => {return [];} },
```

Comment:
i would be tempted to say it must also be mandatory, at least as model object from outside.